src/reference/constellation/index.pug

```diff
@@ -0,0 +1,160 @@
+extends ../reference_layout.pug
+
+block title
+    | Constellation Reference | Beam Developers
+
+block navbar
+    .navbar-wrapper
+        .container
+            nav.navbar.navbar-inverse: +navbar()
+            h1 Constellation Reference
+block menu
+    .hidden-print.rest-sidebar(role='complementary')
+        nav
+            ul.nav.nav-stacked
+                li: a(href='#introduction') Introduction
+                li: a(href='#session') Session
+                li: a(href='#protocol') Protocol
+                li: a(href='#packets') Packets
+                li: a(href='#errors') Error Codes
+                li: a(href='#messages') Messages
+                li: a(href='#events') Events
+block reference
+    .page-header
+        h2#introduction Introduction
+    p.
+        Constellation is the new daemon whose purpose is to unite all stateful function of the website, such as liveloading. This involves:
+    ul
+        li Deprecating the loved/hated Sails socket on realtime.beam.pro and using good ol’ HTTP (well, HTTP/2) for API calls. This drastically simplifies many parts of our stack and debugging.
+        li Creating a new websocket service on constellation.beam.pro which handles liveloading and other statefulness.
+
+    p.
+        The current liveloading socket on realtime.beam.pro will remain in place until #[strong October 1st], at which point it will be removed from Beam.
+
+    h2#session Session
+    p.
+        Users authenticate to Constellation by sending their existing cookies in their socket handshake. Constellation will unseal them, like Hapi does, and automatically create a context for that user. No external API endpoints needed.
+    p.
+        Users remain connected to Constellation throughout their session on the site. During the session, they may subscribe to events happening on the site and get notifications when those events occur.
+
+    h2#protocol Protocol
+    p.
+        Constellation is based on JSON-RPC with additional support for an event system. Initially a user handshakes with #[strong constellation.beam.pro]. Aside from the usual handshake headers, following headers may be passed:
+    ul
+        li #[strong cookie]: may contain the Beam session cookie. Its presence will cause Constellation to attempt to authenticate the user.
+        li #[strong authorization]: may contain and OAuth Bearer token to authenticate with for 3rd party apps, rather than using a cookie. This will indicate to Constellation that the user is a bot.
```

Comment:
sorry for the upcoming nitpicking - "contain **and** OAuth Bearer" should be "contain **an** OAuth Bearer"